pcdsdevices/rtds_ebd.py

```diff
@@ -44,6 +46,27 @@ def _do_move(self, state):
             self.out_cmd.put(1)
 
 
+class RTDSX0ThreeStage(BaseInterface, LightpathMixin):
+    """Three stages X,Y,Z, for solid drilling experiments"""
+
+    mmsx = Cpt(BeckhoffAxis, ':MMS:X', kind='normal')
+    mmsy = Cpt(BeckhoffAxis, ':MMS:Y', kind='normal')
+    mmsz = Cpt(BeckhoffAxis, ':MMS:Z', kind='normal')
+
+    # Determine if the y stage is at it's outer limit switch
+    open_limit = Cpt(EpicsSignalRO, ':MMS:Y.HLS', kind='normal',
+                     doc='Reads if the y-stage is at its outer limit.')
+    lightpath_cpts = ['open_limit']
+
+    def calc_lightpath_state(self, open_limit=None):
+        # Logic, calculations using open_limit for y-stage
+        status = LightpathState(
+            inserted=not open_limit, removed=open_limit,
```

Comment:
This matches the fast fault logic, so this is probably appropriate. The only way this can fail is with a fake "inserted" result with the FFO overridden, right? As in, if the stage is "almost out" and the lightpath says "I'm blocking beam!" It's probably ok.

Reply:
right, this is encoded relatively such that the position is estimated only as a product of the lead screw and [expected] rotation. We've had some issues with this stage where it can stall due to some combination of unexpected friction in the drive and limited current draw observed in the testing phase. There's probably one or two other things contributing to this too. All that to say, the only reasonable "100% safe" state I can think to employ is the outer limit switch being hit.